internal/plugins/scorecard/init.go

```diff
@@ -0,0 +1,259 @@
+}
+
+// kustomizationJSON6902Patch holds path and target data to write a patchesJson6902 list in a kustomization.yaml.
+type kustomizationJSON6902Patch struct {
+	Path   string
+	Target registry.DefinitionKey
+}
+
+// generate scaffolds kustomize bundle bases and a kustomization.yaml.
+// TODO(estroz): refactor this to be testable (in-mem fs) and easier to read.
+func generate(operatorName, testImageTag, outputDir string) error {
+
+	kustomizationValues := scorecardKustomizationValues{}
+
+	// Config bases.
+	basesDir := filepath.Join(outputDir, "bases")
+	if err := os.MkdirAll(basesDir, 0755); err != nil {
+		return err
+	}
+
+	configBase := newScorecardConfigurationBase()
+	b, err := yaml.Marshal(configBase)
+	if err != nil {
+		return fmt.Errorf("error marshaling default config: %v", err)
+	}
+	relBasePath := filepath.Join("bases", scorecard.ConfigFileName)
+	basePath := filepath.Join(basesDir, scorecard.ConfigFileName)
+	if err := ioutil.WriteFile(basePath, b, 0666); err != nil {
+		return fmt.Errorf("error writing default scorecard config: %v", err)
+	}
+	kustomizationValues.ResourcePaths = append(kustomizationValues.ResourcePaths, relBasePath)
+	scorecardConfigTarget := registry.DefinitionKey{
+		Group:   v1alpha3.SchemeGroupVersion.Group,
+		Version: v1alpha3.SchemeGroupVersion.Version,
+		Kind:    v1alpha3.ConfigurationKind,
+		Name:    scorecardConfigName,
+	}
+
+	// Config patches.
+	patchesDir := filepath.Join(outputDir, "patches")
+	if err := os.MkdirAll(patchesDir, 0755); err != nil {
+		return err
+	}
+
+	// Basic scorecard tests patch.
+	basicPatch := newBasicScorecardConfigurationPatch(operatorName, testImageTag)
+	b, err = yaml.Marshal(basicPatch)
+	if err != nil {
+		return fmt.Errorf("error marshaling basic patch config: %v", err)
+	}
+	basicPatchFileName := fmt.Sprintf("basic.%s", scorecard.ConfigFileName)
+	if err := ioutil.WriteFile(filepath.Join(patchesDir, basicPatchFileName), b, 0666); err != nil {
+		return fmt.Errorf("error writing basic scorecard config patch: %v", err)
+	}
+	kustomizationValues.JSONPatches = append(kustomizationValues.JSONPatches, kustomizationJSON6902Patch{
+		Path:   filepath.Join("patches", basicPatchFileName),
+		Target: scorecardConfigTarget,
+	})
+
+	// OLM scorecard tests patch.
+	olmPatch := newOLMScorecardConfigurationPatch(operatorName, testImageTag)
+	b, err = yaml.Marshal(olmPatch)
+	if err != nil {
+		return fmt.Errorf("error marshaling OLM patch config: %v", err)
+	}
+	olmPatchFileName := fmt.Sprintf("olm.%s", scorecard.ConfigFileName)
+	if err := ioutil.WriteFile(filepath.Join(patchesDir, olmPatchFileName), b, 0666); err != nil {
+		return fmt.Errorf("error writing default scorecard config: %v", err)
+	}
+	kustomizationValues.JSONPatches = append(kustomizationValues.JSONPatches, kustomizationJSON6902Patch{
+		Path:   filepath.Join("patches", olmPatchFileName),
+		Target: scorecardConfigTarget,
+	})
+
+	// Write a kustomization.yaml to outputDir if one does not exist.
```

Comment:
We might want to consider writing a marker into the kustomization file if we think we might append to the set of tests during `create api` (e.g. to add a boilerplate kuttl test per API).

Reply:
We should, and perhaps we can just add that in this PR and use it later.

Reply:
Right. And if it goes unused, :man_shrugging: